### DISCUSSION TITLE
How do I conditionally  hide a relation manager based on record field?

### DISCUSSION THREAD

**Discussion by BenQoder** [filamentphp/filament#2179] [Help] [answered]
I would love to know how I would hide a relation manager if  a field is true or false?

for instance I don't want products relations to display if a user is_business is false.

**Comment by danharrin** [comment, marked as answer]
```php
public static function canViewForRecord(Model $ownerRecord): bool
{
    return ! $ownerRecord->is_business;
}
```

**Reply by incloudsistemas** [reply]
Is it possible to do this on live? Ex: when selecting is_business in the owner's form, hide relation manager without reloading the page and vice versa?

**Reply by danharrin** [reply]
`$ownerRecord` is the latest saved version of the record, not the current form data that is unsaved

**Reply by incloudsistemas** [reply]
maybe I expressed badly, what I would like is that on the edit page, when checking a field in the $form I can hide the relationship table. It's possible?

**Reply by danharrin** [reply]
The relation manager currently has no way of accessing the data from the parent form as they are different Livewire components

**Reply by danharrin** [reply]
So no, the user would have to save the form for this to work

**Comment by BenQoder** [comment]
It Works, Thanks A Lot

**Comment by eugenefvdm** [comment]
Could someone please tell me where it's supposed to go in this scenario:

A user has many invoices
A user may have subscriptions.

I only want to conditionally display subscriptions.

I tried adding the above to the SubscriptionsRelationManager but I get:

```
Method 'App\Filament\Resources\UserResource\RelationManagers\SubscriptionsRelationManager::canViewForRecord()' is not compatible with method 'Filament\Resources\RelationManagers\RelationManager::canViewForRecord()'.intelephense(1038)
```

I also tried adding it to the UserResource, the Subscription Policy, the Subscription Model. Basically I'm at a loss where I'm supposed to add the code.

**Reply by danharrin** [reply]
You add it to the Relation Manager, but you haven't copied the method from the base class correctly. The error tells you what is wrong. You need to exactly copy the typehint from the base class, and make sure you import the correct classes if there are any. Good IDEs do this for you.

**Reply by eugenefvdm** [reply]
Thanks man it appears I didn't import `Model` but composer update seemed to also have fixed the problem.
